### ISSUE TITLE
Rollup lost an entire piece of code

### ISSUE THREAD

**Issue by arguiot** [rollup/rollup#3533]
- Rollup Version: v2.7.3
- Operating System (or Browser): macOS Catalina 10.15.5 on Safari
- Link to reproduction (IMPORTANT, read below):

Here is the REPL [link](https://rollupjs.org/repl/?version=2.7.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwbGlzdCUyMCUzRCUyMCU1QjElNUQlM0IlNUNuJTJGJTJGJTIwVGhpcyUyMHdvcmtzJTVDbmxpc3QuZm9yRWFjaChwJTIwJTNEJTNFJTIwJTdCJTVDbiU1Q3Rjb25zb2xlLmxvZyhwKSUzQiU1Q24lN0QpJTNCJTVDbiU1Q24lMkYlMkYlMjBUaGlzJTIwZG9lcyUyMG5vdCUyMHdvcmslNUNubGlzdC5mb3JFYWNoKHAlMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwbGlzdC5mb3JFYWNoKHBwJTIwJTNEJTNFJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKHAlMkMlMjBwcCklM0IlNUNuJTVDdCU3RCklNUNuJTdEKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzbSUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE).

I didn't past all my code as it's private, but this roughly what happened. Here are screenshots of the result I'm getting:
<img width="567" alt="Capture d’écran 2020-05-03 à 19 15 27" src="https://user-images.githubusercontent.com/18022260/80928449-85df1800-8d72-11ea-8352-86641284e26e.png">

I managed to find a workaround by not having a `list` constant:
<img width="819" alt="Capture d’écran 2020-05-03 à 19 16 55" src="https://user-images.githubusercontent.com/18022260/80928474-ab6c2180-8d72-11ea-8148-72b4a682f9f7.png">

### Expected Behavior
Rollup should always keep the code when not certain if it's accessible or not. So it doesn't lose any important code

### Actual Behavior
Inside a class, Rollup lost an entire piece of code for no reason. Maybe Rollup thought that the code wasn't accessible.

> If you ever need the real code to fix the issue, please tell me here, so I can share the code by email.

**Comment by lukastaegert** [comment]
If code is missing, then Rollup did not find a side-effect. The important part would be to see the original code of the `monitor` function without your fix so that I can understand what side-effect was overlooked.

**Comment by arguiot** [comment]
Concerning the original code of the monitor function, here it is:
```js
monitor() {
  const persons = [this.bob, this.alice, this.charlie, this.david]
  const list = persons.filter(person => person.isPark)
  list.forEach(p => {
            list.forEach(pp => {
                if (p == pp) {
                    return
                }
                const diffX = p.x - pp.x;
                const diffY = p.y - pp.y;

                const dist = Math.sqrt(diffX * diffX + diffY * diffY);
                if (dist <= 40) {
                    this.contact(p, pp)
                }
            })
        })
}
```

**Comment by lukastaegert** [comment]
Thanks, I was able to reduce it to a minimal reproduction:

https://rollupjs.org/repl/?version=2.7.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwbGlzdCUyMCUzRCUyMCU1QjElNUQlM0IlNUNuJTJGJTJGJTIwVGhpcyUyMHdvcmtzJTVDbmxpc3QuZm9yRWFjaChwJTIwJTNEJTNFJTIwJTdCJTVDbiU1Q3Rjb25zb2xlLmxvZyhwKSUzQiU1Q24lN0QpJTNCJTVDbiU1Q24lMkYlMkYlMjBUaGlzJTIwZG9lcyUyMG5vdCUyMHdvcmslNUNubGlzdC5mb3JFYWNoKHAlMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwbGlzdC5mb3JFYWNoKHBwJTIwJTNEJTNFJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKHAlMkMlMjBwcCklM0IlNUNuJTVDdCU3RCklNUNuJTdEKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzbSUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by lukastaegert** [comment]
Thanks again, pending fix at #3539
